rdt-app/src/main/java/io/ona/rdt_app/fragment/RDTJsonFormFragment.java

```diff
@@ -51,6 +71,25 @@ public void onClick(View v) {
         });
     }
 
+    public void setNextButtonState(View rootView, boolean buttonEnabled) {
+        Button button = rootView.findViewById(com.vijay.jsonwizard.R.id.next_button);
```

Comment:
In this method, you can test if the button is already in the desired state and if so, avoid doing the processing that follows.